### PR TITLE
[ADD] product_template_has_one_variant

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
LGTM. 

Improvment possible : overload the product template search view, adding two new filter : 
- Has one variant
- Has many variant

**Comment by rousseldenis** [comment]
@OCA/board It seems that runboat instances are a little bit sick. None of them can start

**Review by lmignon — COMMENTED** [review]
Thank you @rousseldenis 
LGTM and useful for some specific usecase where we check if we've more than 1 variant by template

File: product_template_has_one_variant/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        index=True,
+    )
+
+    @api.depends("product_variant_ids")
+    def _compute_has_one_variant(self):
+        templates = self.filtered(
+            lambda template: len(template.product_variant_ids) == 1
+        )
+        templates.update({"has_one_variant": True})
+        (self - templates).update(
```
Comment (simahawk) [inline product_template_has_one_variant/models/product_template.py:23]: are we sure this subtraction won't work on hundreds of records?

File: product_template_has_one_variant/tests/test_product_template.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestProductTemplateHasOneVariant(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.ProductAttribute = cls.env["product.attribute"]
```
Comment (simahawk) [inline product_template_has_one_variant/tests/test_product_template.py:12]: tracking_disable please?

**Comment by rousseldenis** [comment]
@OCA/board @OCA/core-maintainers  Can someone restart runboat as it seems to be dead?

File: product_template_has_one_variant/models/product_template.py (hunk 1 above)
Comment (rousseldenis) [reply product_template_has_one_variant/models/product_template.py:23]: Not sure case will happen as this triggers on product_variant_ids modification

**Comment by rousseldenis** [comment]
> LGTM.
> 
> Improvment possible : overload the product template search view, adding two new filter :
> 
> * Has one variant
> * Has many variant

@legalsylvain The first filter is already there :-)

File: product_template_has_one_variant/tests/test_product_template.py (hunk 2 above)
Comment (rousseldenis) [reply product_template_has_one_variant/tests/test_product_template.py:12]: Done

**Review by legalsylvain — APPROVED** [review]
OK !

**Comment by rousseldenis** [comment]
@simahawk Could you update review?

**Review by FrancoMaxime — APPROVED** [review]
Code review and functionally tested

**Review by lmignon — APPROVED** [review]
Code review and functional tests

File: product_template_has_one_variant/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+On product template list views, you can filter products that has only one product
```
Comment (baimont) [inline product_template_has_one_variant/readme/USAGE.rst]:
```suggestion
On product template list views, you can filter products that have only one product
```

**Review by baimont — APPROVED** [review]

**Review by simahawk — APPROVED** [review]
LG, one typo in the readme

**Pull request merged**